main_test.go

```diff
@@ -0,0 +1,194 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+)
+
+func testRun(t *testing.T, wasmFile string, flags ...string) ([]byte, int) {
+	var logs bytes.Buffer
+	output := io.MultiWriter(testLogger(t), &logs)
+	exitCode := run(append([]string{"go_js_wasm_exec", wasmFile, "-test.v"}, flags...), output, testFlagSet(t))
+	return logs.Bytes(), exitCode
+}
+
+func testFlagSet(t *testing.T) *flag.FlagSet {
+	return flag.NewFlagSet("wasmbrowsertest", flag.ContinueOnError)
+}
+
+type testWriter struct {
```

Comment:
This file might benefit from a bit of code layout re-arrangement. I'd expect the main driver `Test` function to be written first, and then the struct and the `Write` method. And lastly, the remaining helper functions.

Reply:
Fair, rearranged in 0c6425b